gobblin-core/src/main/java/org/apache/gobblin/converter/avro/JsonElementConversionWithAvroSchemaFactory.java

```diff
@@ -165,7 +166,10 @@ public EnumConverter(String fieldName, boolean nullable, String sourceType, Sche
 
     @Override
     Object convertField(JsonElement value) {
-      return new GenericData.EnumSymbol(this.schema, value.getAsString());
+      String valueString = value.getAsString();
+      Validate.isTrue(this.enumSet.contains(valueString),
```

Comment:
most of our check is done with guava `Preconditions`. Replace `Validate.isTrue` with `Preconditions` to be consistent and avoid introducing a new dependency?

Reply:
No problem. The code already has a dependency on commons lang3 (I checked the `build.gradle` of the `gobblin-core` module) but I do agree regarding consistency.

One more thing: I had to make a tiny change to line #162 in `JsonElementConversionWithAvroSchemaFactory.java` because the code wasn't picking the enum name correctly. It was retrieving the type name instead, which is always `enum`. So, now the exception message reads as follows
```
PURPLE is not one of the valid symbols for the org.apache.gobblin.test.Colors enum: [RED, GREEN, BLUE]
```